### PR TITLE
Relax version requirement for rubyzip

### PR DESCRIPTION
**Pull request by ioanatia**
closes https://github.com/jruby/warbler/issues/490

I haven't actually test that it works with the latest version of rubyzip, but looking at the usage of the gem it looks like it should work.

Not sure whether the CI job will actually install the latest version of the gem or not.

### CODE REVIEW

File: warbler.gemspec [hunk 1]
```diff
@@ -31,7 +31,7 @@ bundle up all of your application files for deployment to a Java environment.}
   gem.add_runtime_dependency 'rake', ['>= 10.1.0']
   gem.add_runtime_dependency 'jruby-jars', ['>= 9.0.0']
   gem.add_runtime_dependency 'jruby-rack', ['>= 1.1.1', '< 1.3']
-  gem.add_runtime_dependency 'rubyzip', ['~> 1.0', '< 1.4']
+  gem.add_runtime_dependency 'rubyzip', ['>= 1.0.0', '<= 2.3.0']
```
Comment (olleolleolle) [inline warbler.gemspec:34]: How about not bounding?

```suggestion
  gem.add_runtime_dependency 'rubyzip', '>= 1.0.0'
```

**Review by olleolleolle — APPROVED** [review]
Thanks for contributing this change!

**Comment by olleolleolle** [comment]
The test failures on 9.2.0.0 _seem_ perhaps to be about that, on 9.2 (whereas 9.1 does not have that).

```
  1) Warbler::Jar in a jar project with a .gemspec compiles included gems when compile_gems is true
     Failure/Error: file_list(%r{gems.*\.class$}).size.should >= 80 # depending on RubyZip version
       expected: >= 80
            got:    45
     # ./spec/warbler/jar_spec.rb:186:in `block in (root)'
```

**Comment by ioanatia** [comment]
I'll check the test failures.

**Comment by ioanatia** [comment]
CI failed with:
```
ERROR:  Could not find a valid gem 'bundler' (< 2), here is why:
          Unable to download data from https://rubygems.org/ - Received fatal alert: handshake_failure (https://api.rubygems.org/specs.4.8.gz)
The command "gem install bundler -v '< 2'" failed and exited with 2 during .
```

I'll try to trigger a rebuild with a `git commit --amend --no-edit` and `git push -f`.

**Comment by olleolleolle** [comment]
Someone _had_ an issue a bit like this, here: https://github.com/rubygems/rubygems/issues/4150 

@headius Are we committed to keeping 9.0.5.0 in the matrix still?

**Comment by ioanatia** [comment]
@olleolleolle @headius I removed 9.0.5.0 from .travis.yml and now the tests are green.
Are there any other changes that need to be made if 9.0.5.0 is removed?

**Comment by olleolleolle** [comment]
@ioanatia https://github.com/jruby/warbler#warbler--- the README has a promise of "9.0.0.0 and up", which I believe to be not the case anymore if we fail to run our suite on that range. Could we say, @headius, that we no longer support the < 9.1?

**Comment by olleolleolle** [comment]
@ioanatia Awesome that you edited the CI, this was new information, good!

**Comment by ioanatia** [comment]
> the README has a promise of "9.0.0.0 and up", which I believe to be not the case anymore if we fail to run our suite on that range

Thanks, the README is updated.
@olleolleolle if we go with this approach, should we also add `9.0.0.0` to `.travis.yml` or having `jruby-9.1.17.0` is sufficient?

**Comment by olleolleolle** [comment]
@ioanatia I think latest 9.1 is sufficient for the CI.

**Comment by ioanatia** [comment]
AFAICS jruby 9.0 is supposed to be compatible with Ruby 2.2, but support for Ruby 2.2 has [ended in June 2018](https://www.ruby-lang.org/en/news/2018/06/20/support-of-ruby-2-2-has-ended/).
It might be okay to move forward and remove support for 9.0 in newer versions of warbler @olleolleolle @headius

**Comment by deivid-rodriguez** [comment]
Technically jruby 9.1 is Ruby 2.3 compatible and support for Ruby 2.3 has also ended. In addition to that, bundler 2 does not really work on jruby 9.1 due to some issues in jruby fixed in the 9.2 series (see https://github.com/ruby/setup-ruby/issues/108).

So maybe it's ok to drop support for jruby 9.1 too in the next version.

**Comment by deivid-rodriguez** [comment]
Also I just installed `jruby 9.1` through `ruby-build` and I got this message: 

> WARNING: jruby-9.1.17.0 is past its end of life and is now unsupported.
> It no longer receives bug fixes or critical security updates.

It seems official that jruby 9.1 is out of support, so I think it's ok to move on.

**Review by olleolleolle — APPROVED** [review]
Thanks all, for chiming in! Merging now!

**Comment by olleolleolle** [comment]
@ioanatia Thanks for the strong momentum and follow-through!

**Comment by ioanatia** [comment]
Nice! Thanks for merging this @olleolleolle - any chance to have a new release of this gem soon?

**Comment by olleolleolle** [comment]
@ioanatia That's a little longer process, but with @deivid-rodriguez working on a GitHub Actions CI change, and we are at a green build, so if you want to support the  process, please investigate what information we have about performing releases. I am new to that process, and am documenting my steps in a coordination Issue linked here.

**Comment by deivid-rodriguez** [comment]
I found that this PR might've introduced an issue.

Before, the `rake integration` task was failing, but at least completing and showing "Passed: 1, Failed: 3". See the end of the log here: https://travis-ci.org/github/jruby/warbler/jobs/771640167.

With this PR it's failing to even start with the error: "Error installing /home/travis/.m2/repository/rubygems/rubyzip/2.3.0/rubyzip-2.3.0.gem: rubyzip requires Ruby version >= 2.4.".

The "solution" seems to (again) to drop support for jruby 9.1 as well, so I'm going to propose that as a separate PR.

And that surfaces yet another issue because the `rake integration` job is shown as green, where it's actually broken. I'll try to propose yet another PR to fix that as well.

**Pull request merged**